lualib/ngx/ssl/session/ticket/key_rotation.lua

```diff
@@ -283,25 +238,68 @@ function _M.init(opts)
     time_slot = opts.key_rotation_period
     memc_key_prefix = opts.memc_key_prefix
 
-    shdict_name = opts.cache_shdict_name
-    shm_cache_pos_ttl = opts.shm_cache_positive_ttl
-    shm_cache_neg_ttl = opts.shm_cache_negative_ttl
-    disable_shm_cache = opts.disable_shm_cache
-    locks_shdict_name = opts.locks_shdict_name
+    local shdict_name = opts.cache_shdict_name
+    local shm_cache_pos_ttl = opts.shm_cache_positive_ttl
+    local shm_cache_neg_ttl = opts.shm_cache_negative_ttl
+    local disable_shm_cache = opts.disable_shm_cache
+    local locks_shdict_name = opts.locks_shdict_name
 
-    memc_host = opts.memc_host
-    memc_port = opts.memc_port
-    memc_timeout = opts.memc_timeout
-    memc_conn_pool_size = opts.memc_conn_pool_size
-    memc_fetch_retries = opts.memc_fetch_retries
-    memc_fetch_retry_delay = opts.memc_fetch_retry_delay
+    local memc_host = opts.memc_host
+    local memc_port = opts.memc_port
+    local memc_timeout = opts.memc_timeout
+    local memc_conn_pool_size = opts.memc_conn_pool_size
+    local memc_fetch_retries = opts.memc_fetch_retries
+    local memc_fetch_retry_delay = opts.memc_fetch_retry_delay
 
-    memc_conn_max_idle_time = opts.memc_conn_max_idle_time
+    local memc_conn_max_idle_time = opts.memc_conn_max_idle_time
 
     local frandom = assert(io.open("/dev/urandom", "rb"))
     fallback_random_key = frandom:read(48)
     frandom:close()
 
+    nkeys = floor(ticket_ttl / time_slot) + 2
+    do
+        local meta_shdict = require "resty.shdict.simple"
+        meta_shdict_set, meta_shdict_get = meta_shdict.gen_shdict_methods{
+            dict_name = shdict_name,
+            debug_logger = dlog,
+            warn_logger = warn,
+            error_logger = error_log,
+            positive_ttl = shm_cache_pos_ttl,
+            negative_ttl = shm_cache_neg_ttl,
+        }
+    end
+
+    do
+        local memc_shdict = require "resty.memcached.shdict"
+        fetch_key_from_memc = memc_shdict.gen_memc_methods{
+            tag = "ticket_memc",
+
+            debug_logger = dlog,
+            warn_logger = warn,
+            error_logger = error_log,
+
+            locks_shdict_name = locks_shdict_name,
+
+            disable_shdict = disable_shm_cache,
+
+            shdict_set = meta_shdict_set,
+            shdict_get = meta_shdict_get,
+
+            memc_host = memc_host,
+            memc_port = memc_port,
+            memc_timeout = memc_timeout,
+            memc_conn_pool_size = memc_conn_pool_size,
+            memc_fetch_retries = memc_fetch_retries,
+            memc_fetch_retry_delay = memc_fetch_retry_delay,
+
+            memc_conn_max_idle_time = memc_conn_max_idle_time,
+
+            memc_store_retries = 0,
+            memc_store_retry_delay = 0,
+        }
+    end
+
     local ctx = {}
```

Comment:
@xiewenhui Yeah, you are right:) Seems useless only in a few functions like `dlog`, but it seems doesn't matter.